CARET_trace/src/hooked_trace_points.cpp

```diff
@@ -233,6 +237,29 @@ int dds_write_impl(void * wr, void * data, long tstamp, int action)  // NOLINT
   return dds_return;
 }
 
+// for cyclonedds
+// bind : &ros_message -> source_timestamp
+// cspell: ignore ddsi, serdata, dinp
+int dds_writecdr_impl(void * wr, void * xp, struct ddsi_serdata * dinp, bool flush)  // NOLINT
```

Comment:
Is it possible to hook dds_witecdr instead of dds_writecdr_impl? I think dds_witecdr is better because it is api function and less likely to be changed in the future.

Reply:
Since dds_writecdr is an api function, it is preferable to hook this function to output trace points if possible.
However, when dds_writecdr is hooked, source_timestamp cannot be obtained, so dds_writecdr_impl must be hooked.

Reply:
You are right. I confirmed that argument of dds_writecdr does not contain source_timestamp.